File: mobile/src/main/java/tr/xip/prayertimes/util/LocationsList.java
package tr.xip.prayertimes.util;

import java.util.ArrayList;

import tr.xip.prayertimes.api.objects.Location;

public class LocationsList extends ArrayList<Location> {

    public Location findById(int id) {
        if (size() == 0) {
            return null;
        }

        if (id == -1) {
            return get(0);
        }

        Location result = null;
        for (Location l : this) {
            if (l.getDatabaseId() == id) {
                result = l;
                break;
            }
        }
        return result;
    }
}

File: mobile/src/main/kotlin/tr/xip/prayertimes/util/LocationsList.kt
package tr.xip.prayertimes.util

import java.util.ArrayList

import tr.xip.prayertimes.model.Location

class LocationsList : ArrayList<Location>() {

    fun findById(id: Int): Location? {
        if (size == 0) {
            return null
        }

        if (id == -1) {
            return get(0)
        }

        var result: Location? = null
        for (item in this) {
            if (item.databaseId == id) {
                result = item
                break
            }
        }

        return result
    }
}
